DM-25477: Provide release notes for 20.0.0 (#157)
lsst/pipelines_lsst_io: releases/note-source/v20_0_0.rst

The `Conda <https://conda.io>`_ package management system, together with the `Conda-Forge <https://conda-forge.org>`_ package collection, is now used to provide all compilers and most third-party dependencies.
This substantially simplifies :ref:`part-installation` and makes it easier to integrate the Science Pipelines with other tools, while also facilitating quicker updates to third-party code during development.

For more information, refer to `community.lsst.org <https://community.lsst.org/t/conda-updates-implementing-the-switch-to-conda-forge-conda-compilers-rfc-679/4127>`_.
:jirab:`RFC-679, DM-24232`

.. _release-v20-0-0-fits-headers:

All FITS headers are now stored in uppercase
^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^

The version of `CFITSIO`_ used in this release requires that all header keys be written in uppercase to better comply with the `FITS standard`_.
Before this release, some Science Pipelines code would store information in FITS headers in lowercase.
This has the consequence that:

- If user code attempts to write a lowercase header to FITS, it will be automatically converted to uppercase, and a warning will be issued;

> - If user code attempts to read FITS data which includes lowercase headers (for example, because it was written by an earlier Pipelines release), they will automatically be converted to uppercase, and a warning will be issued.

Reviewer: ```suggestion
- If user code attempts to read FITS data which includes lowercase headers (for example, because it was written by an earlier Pipelines release), they will automatically be converted to uppercase, and an informational message may be logged.
```

:jirab:`RFC-640, RFC-686, DM-21989, DM-21991, DM-24376, DM-24570, DM-24618`

.. _CFITSIO: https://heasarc.gsfc.nasa.gov/fitsio/
.. _FITS standard: https://fits.gsfc.nasa.gov/fits_standard.html

.. _release-v20-0-0-collections:

Changes to collections, runs, and ``pipetask`` arguments in Generation 3 middleware
^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^

> The way in which collections behave in the Generation 3 middleware have been substantially overhauled.

Reviewer: ```suggestion
The way in which collections behave in the Generation 3 middleware has been substantially overhauled.
```

In particular:

- a “run” is now a special type of collection, rather than an entity that is associated with a collection;
- the only kind of collection we had before is now called a “tagged” collection;
- we now also have “chained” collections, which are simply an ordered list of other collections to be searched.

This changes some prominent interfaces, including both the arguments used to construct a `~lsst.daf.butler.Butler` and the command-line arguments to the ``pipetask`` tool.

For more information, see `this community.lsst.org posting <https://community.lsst.org/t/changes-to-collections-runs-and-pipetask-arguments-in-gen3-middleware/4078>`_ and the :ref:`daf_butler <lsst.daf.butler>` documentation.
:jirab:`RFC-663, DM-21849`

.. _release-v20-0-0-lsstsw-envconfig:

Changes to building using lsstsw
^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^

The initialization scripts used by the :ref:`lsstsw <install-lsstsw>` system, used by some developers to install and update the Science Pipelines, have been renamed.
Where previously you sourced :file:`setup.sh` (or :file:`setup.csh`), you should now source :file:`envconfig.sh` (or :file:`envconfig.csh`).

> :jirab:`DM-23098`

Reviewer: Perhaps add "This helps distinguish this script from the `setup` command provided by `eups`."
